zend_abstract_interface/properties/tests/php7-8/properties.cc

```diff
@@ -0,0 +1,134 @@
+extern "C" {
+#include "zai_sapi/zai_sapi.h"
+#include "properties/properties.h"
+#include "functions/functions.h"
+}
+
+#include <catch2/catch.hpp>
+#include <cstring>
+
+#define TEST(name, code) TEST_CASE(name, "[zai properties]") { \
+        REQUIRE(zai_sapi_spinup()); \
+        ZAI_SAPI_TSRMLS_FETCH(); \
+        ZAI_SAPI_ABORT_ON_BAILOUT_OPEN() \
+        REQUIRE(zai_sapi_execute_script("./stubs/classes.php")); \
+        { code } \
+        ZAI_SAPI_ABORT_ON_BAILOUT_CLOSE() \
+        zai_sapi_spindown(); \
+    }
+
+static inline zval *zai_read_property_direct_literal(zend_class_entry *ce, zend_object *obj, const char *name, int name_len) {
+    zend_string *name_str = zend_string_init(name, name_len, 0);
+    zval *val = zai_read_property_direct(ce, obj, name_str);
+    zend_string_release(name_str);
```

Comment:
Do you think it's worth using the `ZSTR_ALLOCA_ALLOC` API here?

Reply:
yes, as we move it to public API, yes.